Triple quote in cypher string + multiline = error

Hello it's me again

### ArcadeDB Version:

` ArcadeDB Server v24.6.1 (build fbc1c77c7e1c52197abdd8661e5b34749babf3d6/1721708610769/main)`

### OS and JDK Version:

`unning on Linux 6.5.0-41-generic - OpenJDK 64-Bit Server VM 11.0.24
`

### Expected behavior

working query

### Actual behavior

Erroring query (triple (or more) quotes only)
```CYPHER
CREATE (n:CHUNK { text: ' double 
        line"""'}) RETURN ID(n)
```

Working query : 

```CYPHER
CREATE (n:CHUNK { text: ' single line"""   '}) RETURN ID(n)
```

### Error: 



```txt

oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]: 2024-10-21 14:30:06.127 INFO  [ArcadeDBServer] <ArcadeDB_0> Studio web tool available at http://192.168.14.29:2480 Script compilation FAILED g.addV('CHUNK').as('n').property(single, 'te>
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         line"""""").select('n').project('ID(n)').by(__.id()) took 143ms {}
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]: org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]: Script1.groovy: 2: Unexpected input: '('n').property(single, 'text', """ double\n        line"""""' @ line 2, column 16.
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:            line"""""").select('n').project('ID(n)').by(__.id())
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:                   ^
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]: 1 error
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.codehaus.groovy.control.ErrorCollector.failIfErrors(ErrorCollector.java:292)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.codehaus.groovy.control.ErrorCollector.addFatalError(ErrorCollector.java:148)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.AstBuilder.collectSyntaxError(AstBuilder.java:4789)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.AstBuilder.access$100(AstBuilder.java:169)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.AstBuilder$3.syntaxError(AstBuilder.java:4800)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovyjarjarantlr4.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:44)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovyjarjarantlr4.v4.runtime.Parser.notifyErrorListeners(Parser.java:543)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovyjarjarantlr4.v4.runtime.DefaultErrorStrategy.notifyErrorListeners(DefaultErrorStrategy.java:154)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.internal.DescriptiveErrorStrategy.reportNoViableAlternative(DescriptiveErrorStrategy.java:92)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovyjarjarantlr4.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.pathExpression(GroovyParser.java:9907)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.postfixExpression(GroovyParser.java:8282)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.expression(GroovyParser.java:9071)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.commandExpression(GroovyParser.java:9709)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.statementExpression(GroovyParser.java:8241)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.statement(GroovyParser.java:7086)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.scriptStatement(GroovyParser.java:517)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.scriptStatements(GroovyParser.java:416)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.compilationUnit(GroovyParser.java:363)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.AstBuilder.buildCST(AstBuilder.java:243)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.AstBuilder.buildCST(AstBuilder.java:221)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.AstBuilder.buildAST(AstBuilder.java:262)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.Antlr4ParserPlugin.buildAST(Antlr4ParserPlugin.java:58)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.codehaus.groovy.control.SourceUnit.buildAST(SourceUnit.java:257)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:663)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:373)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovy.lang.GroovyClassLoader.lambda$parseClass$2(GroovyClassLoader.java:316)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.codehaus.groovy.runtime.memoize.StampedCommonCache.compute(StampedCommonCache.java:163)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.codehaus.groovy.runtime.memoize.StampedCommonCache.getAndPut(StampedCommonCache.java:154)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:314)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:298)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:258)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine$GroovyCacheLoader.lambda$load$0(GremlinGroovyScriptEngine.java:824)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.concurrent.CompletableFuture.asyncSupplyStage(CompletableFuture.java:1714)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.concurrent.CompletableFuture.supplyAsync(CompletableFuture.java:1937)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine$GroovyCacheLoader.load(GremlinGroovyScriptEngine.java:822)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine$GroovyCacheLoader.load(GremlinGroovyScriptEngine.java:817)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalLoadingCache.lambda$new$0(BoundedLocalCache.java:3117)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.LocalCache.lambda$statsAware$0(LocalCache.java:144)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$16(BoundedLocalCache.java:1968)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1908)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:1966)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:1949)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:113)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.LocalLoadingCache.get(LocalLoadingCache.java:67)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.getScriptClass(GremlinGroovyScriptEngine.java:572)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.eval(GremlinGroovyScriptEngine.java:379)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:233)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.gremlin.ArcadeGremlin.executeStatement(ArcadeGremlin.java:235)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.gremlin.ArcadeGremlin.executeStatement(ArcadeGremlin.java:212)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.gremlin.ArcadeGremlin.execute(ArcadeGremlin.java:73)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.cypher.ArcadeCypher.execute(ArcadeCypher.java:66)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.cypher.query.CypherQueryEngine.command(CypherQueryEngine.java:77)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.database.LocalDatabase.command(LocalDatabase.java:1348)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.server.ServerDatabase.command(ServerDatabase.java:472)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.server.http.handler.PostCommandHandler.executeCommand(PostCommandHandler.java:134)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.server.http.handler.PostCommandHandler.execute(PostCommandHandler.java:113)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.server.http.handler.DatabaseAbstractHandler.execute(DatabaseAbstractHandler.java:100)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.server.http.handler.AbstractServerHttpHandler.handleRequest(AbstractServerHttpHandler.java:127)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at io.undertow.server.Connectors.executeRootHandler(Connectors.java:393)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:859)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1512)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.xnio.XnioWorker$WorkerThreadFactory$1$1.run(XnioWorker.java:1282)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.lang.Thread.run(Thread.java:829)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]: <ArcadeDB_0> Error on command execution (PostCommandHandler)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]: com.arcadedb.exception.CommandParsingException: Error on executing Cypher query
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.cypher.query.CypherQueryEngine.command(CypherQueryEngine.java:80)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.database.LocalDatabase.command(LocalDatabase.java:1348)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.server.ServerDatabase.command(ServerDatabase.java:472)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.server.http.handler.PostCommandHandler.executeCommand(PostCommandHandler.java:134)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.server.http.handler.PostCommandHandler.execute(PostCommandHandler.java:113)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.server.http.handler.DatabaseAbstractHandler.execute(DatabaseAbstractHandler.java:100)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.server.http.handler.AbstractServerHttpHandler.handleRequest(AbstractServerHttpHandler.java:127)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at io.undertow.server.Connectors.executeRootHandler(Connectors.java:393)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:859)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1512)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.xnio.XnioWorker$WorkerThreadFactory$1$1.run(XnioWorker.java:1282)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.lang.Thread.run(Thread.java:829)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]: Caused by: com.arcadedb.exception.CommandExecutionException: Error on executing command
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.gremlin.ArcadeGremlin.execute(ArcadeGremlin.java:139)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.cypher.ArcadeCypher.execute(ArcadeCypher.java:66)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.cypher.query.CypherQueryEngine.command(CypherQueryEngine.java:77)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         ... 13 more
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]: Caused by: javax.script.ScriptException: org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]: Script1.groovy: 2: Unexpected input: '('n').property(single, 'text', """ double\n        line"""""' @ line 2, column 16.
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:            line"""""").select('n').project('ID(n)').by(__.id())
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:                   ^
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]: 1 error
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.eval(GremlinGroovyScriptEngine.java:383)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:233)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.gremlin.ArcadeGremlin.executeStatement(ArcadeGremlin.java:235)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.gremlin.ArcadeGremlin.executeStatement(ArcadeGremlin.java:212)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.arcadedb.gremlin.ArcadeGremlin.execute(ArcadeGremlin.java:73)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         ... 15 more
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]: Caused by: org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]: Script1.groovy: 2: Unexpected input: '('n').property(single, 'text', """ double\n        line"""""' @ line 2, column 16.
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:            line"""""").select('n').project('ID(n)').by(__.id())
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:                   ^
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]: 1 error
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.codehaus.groovy.control.ErrorCollector.failIfErrors(ErrorCollector.java:292)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.codehaus.groovy.control.ErrorCollector.addFatalError(ErrorCollector.java:148)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.AstBuilder.collectSyntaxError(AstBuilder.java:4789)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.AstBuilder.access$100(AstBuilder.java:169)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.AstBuilder$3.syntaxError(AstBuilder.java:4800)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovyjarjarantlr4.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:44)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovyjarjarantlr4.v4.runtime.Parser.notifyErrorListeners(Parser.java:543)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovyjarjarantlr4.v4.runtime.DefaultErrorStrategy.notifyErrorListeners(DefaultErrorStrategy.java:154)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.internal.DescriptiveErrorStrategy.reportNoViableAlternative(DescriptiveErrorStrategy.java:92)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovyjarjarantlr4.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.pathExpression(GroovyParser.java:9907)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.postfixExpression(GroovyParser.java:8282)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.expression(GroovyParser.java:9071)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.commandExpression(GroovyParser.java:9709)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.statementExpression(GroovyParser.java:8241)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.statement(GroovyParser.java:7086)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.scriptStatement(GroovyParser.java:517)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.scriptStatements(GroovyParser.java:416)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.GroovyParser.compilationUnit(GroovyParser.java:363)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.AstBuilder.buildCST(AstBuilder.java:243)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.AstBuilder.buildCST(AstBuilder.java:221)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.AstBuilder.buildAST(AstBuilder.java:262)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.groovy.parser.antlr4.Antlr4ParserPlugin.buildAST(Antlr4ParserPlugin.java:58)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.codehaus.groovy.control.SourceUnit.buildAST(SourceUnit.java:257)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:663)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:373)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovy.lang.GroovyClassLoader.lambda$parseClass$2(GroovyClassLoader.java:316)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.codehaus.groovy.runtime.memoize.StampedCommonCache.compute(StampedCommonCache.java:163)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.codehaus.groovy.runtime.memoize.StampedCommonCache.getAndPut(StampedCommonCache.java:154)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:314)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:298)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:258)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine$GroovyCacheLoader.lambda$load$0(GremlinGroovyScriptEngine.java:824)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.concurrent.CompletableFuture.asyncSupplyStage(CompletableFuture.java:1714)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.concurrent.CompletableFuture.supplyAsync(CompletableFuture.java:1937)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine$GroovyCacheLoader.load(GremlinGroovyScriptEngine.java:822)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine$GroovyCacheLoader.load(GremlinGroovyScriptEngine.java:817)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalLoadingCache.lambda$new$0(BoundedLocalCache.java:3117)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.LocalCache.lambda$statsAware$0(LocalCache.java:144)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$16(BoundedLocalCache.java:1968)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at java.base/java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1908)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:1966)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:1949)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:113)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at com.github.benmanes.caffeine.cache.LocalLoadingCache.get(LocalLoadingCache.java:67)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.getScriptClass(GremlinGroovyScriptEngine.java:572)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.eval(GremlinGroovyScriptEngine.java:379)
oct. 21 14:30:21 cfia-IDEXTEND server.sh[1548715]:         ... 19 more

```

please note that when the query is 

```CYPHER
CREATE (n:CHUNK { text: 'start 
       """ end'}) RETURN ID(n)

```
(Single quotes with 3 double quotes inside)


Error says : 


```txt


oct. 21 14:43:30 cfia-IDEXTEND server.sh[1548715]: Caused by: org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
oct. 21 14:43:30 cfia-IDEXTEND server.sh[1548715]: Script8.groovy: 2: Unexpected input: '('n').property(single, 'text', """start \n       """ end' @ line 2, column 12.
oct. 21 14:43:30 cfia-IDEXTEND server.sh[1548715]:           """ end""").select('n').project('ID(n)').by(__.id())
oct. 21 14:43:30 cfia-IDEXTEND server.sh[1548715]:               ^


```

which translates to 
```java
('n').property(single, 'text', """start \n       """ end""").select('n').project('ID(n)').by(__.id())
```


It seems that something fishy's going on there

--- a/gremlin/src/main/java/org/opencypher/gremlin/translation/groovy/StringTranslationUtils.java
+++ b/gremlin/src/main/java/org/opencypher/gremlin/translation/groovy/StringTranslationUtils.java
@@ -1,0 +1,97 @@
+/*
+ * Copyright (c) 2018-2025 "Neo4j, Inc." [https://neo4j.com]
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opencypher.gremlin.translation.groovy;
+
+import static java.util.stream.Collectors.joining;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public final class StringTranslationUtils {
+    private StringTranslationUtils() {
+    }
+
+    public static String apply(String name, Object... arguments) {
+        String joined = Stream.of(arguments)
+            .map(StringTranslationUtils::toLiteral)
+            .collect(joining(", "));
+        return name + "(" + joined + ")";
+    }
+
+    static String chain(String name, Object... arguments) {
+        return "." + apply(name, arguments);
+    }
+
+    public static String toLiteral(Object argument) {
+        if (argument instanceof List) {
+            return ((List<?>) argument).stream()
+                .map(StringTranslationUtils::toLiteral)
+                .collect(Collectors.joining(", ", "[", "]"));
+        }
+        if (argument instanceof Map) {
+            Map<?, ?> map = (Map<?, ?>) argument;
+            if (map.isEmpty()) {
+                return "[:]";
+            }
+            return map.entrySet().stream()
+                .map(entry -> {
+                    Object key = entry.getKey();
+                    Object value = toLiteral(entry.getValue());
+                    return key + ": " + value;
+                })
+                .collect(Collectors.joining(", ", "[", "]"));
+        }
+        if (argument instanceof Double || argument instanceof Float) {
+            return argument.toString() + "d";
+        }
+        if (argument instanceof String) {
+            return toStringLiteral((String) argument);
+        }
+        if (argument instanceof Double || argument instanceof Float) {
+            return argument.toString() + "d";
+        }
+        if (argument == null) {
+            return "null";
+        }
+        if (argument instanceof Verbatim) {
+            return ((Verbatim) argument).getValue();
+        }
+
+        return argument.toString();
+    }
+
+    private static String toStringLiteral(String argument) {
+        if (argument.contains("\n")) {
+            // Handle multiline strings
+            if (argument.contains("\"\"\"")) {
+                // If the string contains """, use single quotes and escape newlines and quotes
+                return "'" + argument.replaceAll("(['\\\\])", "\\\\$1")
+                                .replaceAll("\n", "\\\\n")
+                                .replaceAll("\r", "\\\\r") + "'";
+            } else {
+                // Use triple quotes for multiline strings, but escape any ending triple quotes
+                String processed = argument.replaceAll("\"\"\"", "\\\\\"\\\\\"\\\\\"");
+                return "\"\"\"" + processed + "\"\"\"";
+            }
+        } else {
+            // For single line strings, use the original approach with single quotes
+            return "'" + argument.replaceAll("(['\\\\])", "\\\\$1") + "'";
+        }
+    }
+
+}

